src/org/thoughtcrime/securesms/LogViewFragment.java

```diff
@@ -87,14 +94,49 @@ public String getLogText() {
 
   public Float getLogTextSize() { return logPreview.getTextSize(); }
 
-  public void setLogTextSize(Float Textsize) {
-    logPreview.setTextSize(TypedValue.COMPLEX_UNIT_PX, Textsize);
+  public void setLogTextSize(Float textSize) {
+    logPreview.setTextSize(TypedValue.COMPLEX_UNIT_PX, textSize);
   }
 
   public void scrollDownLog() { logPreview.setSelection(logPreview.getText().length()); }
 
   public void scrollUpLog() { logPreview.setSelection(0); }
 
+  public boolean saveLogFile() throws NoExternalStorageException {
+
+    File             outputDir   = null;
+    SimpleDateFormat dateFormat  = new SimpleDateFormat("yyyyMMdd-HHmmss");
+    Date             now         = new Date();
+    String           logFileName = "deltachat-log-" + dateFormat.format(now) + ".txt";
+
+    try {
+      outputDir = StorageUtil.getDownloadDir();
+      try {
```

Comment:
there's no need for the nested try catch clause 
catch both exceptions in one place: `... } catch (IOException | NoExternalStorageException e) { ...`